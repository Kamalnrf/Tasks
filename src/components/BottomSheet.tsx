import React, {ReactNode} from 'react'
import {
  KeyboardAvoidingView,
  Modal,
  Platform,
  StyleSheet,
  View,
} from 'react-native'
import {useSafeAreaInsets} from 'react-native-safe-area-context'
import {colors} from '../constants'

const styles = StyleSheet.create({
  rootStyle: {
    flex: 1,
    justifyContent: 'flex-end',
    backgroundColor: colors.black_000000C0,
  },
  dialogStye: {
    padding: 24,
    backgroundColor: colors.black_2C2D2F,
    borderTopLeftRadius: 8,
    borderTopRightRadius: 8,
  },
})

type Props = {
  children: ReactNode
  testId?: string
}

const BottomSheet = ({children, testId}: Props) => {
  const {bottom} = useSafeAreaInsets()

  const KeyboardView = Platform.OS === 'ios' ? KeyboardAvoidingView : View

  return (
    <Modal testID={testId} transparent animationType="slide">
      <View style={[styles.rootStyle, {paddingBottom: bottom}]}>
        <KeyboardView behavior="padding" style={styles.dialogStye}>
          {children}
        </KeyboardView>
      </View>
    </Modal>
  )
}

export default BottomSheet
