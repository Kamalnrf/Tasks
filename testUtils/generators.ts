import faker from 'faker'
import {Tag, Task} from '../src/types'

function generateList<T>(fn: (override?: {}) => T) {
  return (n: number): Array<T> =>
    new Array(n).fill('').reduce((acc, _) => [...acc, fn()])
}

export const generateTag = (override = {}): Tag => ({
  id: faker.random.uuid(),
  name: faker.name.title(),
  ...override,
})

export const generateTags = generateList<Tag>(generateTag)

export const generateTask = (override = {}): Task => ({
  id: faker.random.uuid(),
  endTime: faker.date.recent().toISOString(),
  startTime: faker.date.past().toISOString(),
  tags: generateTags(10),
  title: faker.name.title(),
  ...override,
})
