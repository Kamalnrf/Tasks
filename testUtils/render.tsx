import React from 'react'
import {render as rtlRender} from '@testing-library/react-native'
import {QueryClient, QueryClientProvider} from 'react-query'
import {SafeAreaProvider} from 'react-native-safe-area-context'

const queryClient = new QueryClient()

function render(ui: JSX.Element, {...renderOptions} = {}) {
  function Wrapper({children}: {children: JSX.Element}) {
    return (
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider
          initialMetrics={{
            frame: {x: 0, y: 0, width: 0, height: 0},
            insets: {top: 0, left: 0, right: 0, bottom: 0},
          }}
        >
          {children}
        </SafeAreaProvider>
      </QueryClientProvider>
    )
  }
  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
}

export * from '@testing-library/react-native'

export {render}
