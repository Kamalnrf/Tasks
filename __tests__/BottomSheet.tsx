import React from 'react'
import {Text} from 'react-native'
import BottomSheet from '../src/components/BottomSheet'
import {render} from '../testUtils/render'

describe('Test BottomSheet', () => {
  test('should render bottomsheet with passed children', () => {
    const {toJSON} = render(
      <BottomSheet>
        <Text>Hello</Text>
        <Text>BottomSheet</Text>
      </BottomSheet>,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
