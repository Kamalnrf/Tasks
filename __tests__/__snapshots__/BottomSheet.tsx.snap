// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BottomSheet should render bottomsheet with passed children 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#000000C0",
            "flex": 1,
            "justifyContent": "flex-end",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#2C2D2F",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "padding": 24,
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <Text>
          Hello
        </Text>
        <Text>
          BottomSheet
        </Text>
      </View>
    </View>
  </Modal>
</RNCSafeAreaProvider>
`;
